pkg/template/api/types.go

```diff
@@ -59,4 +59,8 @@ type Parameter struct {
 
 	// Optional: Indicates the parameter must have a value.  Defaults to false.
 	Required bool
+
+	// Optional: Defines the presentation style a UI should use when prompting a user for a value for this parameter,
+	// eg "multiline" or "fileUpload"
```

Comment:
On Mon, Mar 7, 2016 at 8:28 AM, David Eads <notifications@github.com> wrote:

> In pkg/template/api/types.go
> <https://github.com/openshift/origin/pull/7673#discussion_r55202354>:
>
> > @@ -59,4 +59,8 @@ type Parameter struct {
> >
> >  	// Optional: Indicates the parameter must have a value.  Defaults to false.
> >  	Required bool
> > +
> > +	// Optional: Defines the presentation style a UI should use when prompting a user for a value for this parameter,
> > +	// eg "multiline" or "fileUpload"
>
> Remind me again why those are mutually exclusive? Seems like I'd want to
> have both for some things to allow me to choose whether to copy/paste or
> read from file.
>
Sigh, true.  These two in particular are not mutually exclusive.


> It strikes me that this is something I'll be much happier staying out of,
> but I'm betting this ends up delimited. :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7673/files#r55202354>.
>